scapy/arch/linux.py

```diff
@@ -536,46 +575,53 @@ class L2ListenSocket(L2Socket):
     desc = "read packets at layer 2 using Linux PF_PACKET sockets. Also receives the packets going OUT"  # noqa: E501
 
     def send(self, x):
+        # type: (Packet) -> NoReturn
         raise Scapy_Exception("Can't send anything with L2ListenSocket")
 
 
 class L3PacketSocket(L2Socket):
     desc = "read/write packets at layer 3 using Linux PF_PACKET sockets"
 
     def recv(self, x=MTU):
+        # type: (int) -> Optional[Packet]
         pkt = SuperSocket.recv(self, x)
         if pkt and self.lvl == 2:
             pkt.payload.time = pkt.time
             return pkt.payload
         return pkt
 
     def send(self, x):
+        # type: (Packet) -> int
         iff = x.route()[0]
         if iff is None:
             iff = conf.iface
         sdto = (iff, self.type)
         self.outs.bind(sdto)
         sn = self.outs.getsockname()
-        ll = lambda x: x
+        ll = lambda x: x  # type: Callable[[Packet], Packet]
         type_x = type(x)
         if type_x in conf.l3types:
-            sdto = (iff, conf.l3types[type_x])
+            sdto = (iff, conf.l3types.layer2num[type_x])
         if sn[3] in conf.l2types:
-            ll = lambda x: conf.l2types[sn[3]]() / x
+            ll = lambda x: conf.l2types.num2layer[sn[3]]() / x
         if self.lvl == 3 and type_x != self.LL:
             warning("Incompatible L3 types detected using %s instead of %s !",
                     type_x, self.LL)
             self.LL = type_x
         sx = raw(ll(x))
         x.sent_time = time.time()
         try:
-            self.outs.sendto(sx, sdto)
+            return self.outs.sendto(sx, sdto)
```

Comment:
`send` was returning the amount of bytes sent on most platforms but not all of them. Let's make that universal